### PR TITLE
feature/SAAS-1843: Added AWS  ELB SSl Termination plugin and test cases

### PR DESCRIPTION
**Pull request by UmairAjmal**

### CODE REVIEW

File: plugins/aws/elbv2/elbv2SslTermination.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                    helpers.addResult(results, 2,
+                        'Elastic Load Balancer does not have SSL Termination configured',
+                        region, resource);
+                    }
+                }
+
+                cb();
+            });
+
+            rcb();
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2SslTermination.js:74]:
```suggestion
                cb();
            }, function() {
                rcb();
            });
```

File: plugins/aws/elbv2/elbv2SslTermination.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                        'No Load Balancer listeners found',
+                        region, resource);
+                    return cb();
+                }
+
+                for (var l in describeListeners.data.Listeners) {
+                    var listener = describeListeners.data.Listeners[l];
+                    if (listener.Certificates && listener.Certificates.length) {
+                        console.log(listener.Certificates, listener.DefaultActions);
+                        helpers.addResult(results, 0,
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2SslTermination.js:61]:
```suggestion
                    if (listener.Certificates && listener.Certificates.length) {
                        helpers.addResult(results, 0,
```

File: plugins/aws/elbv2/elbv2SslTermination.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                            'Elastic Load Balancer has SSL Termination configured',
+                            region, resource);
+                    } else {
+                    helpers.addResult(results, 2,
+                        'Elastic Load Balancer does not have SSL Termination configured',
+                        region, resource);
+                    }
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2SslTermination.js:68]:
```suggestion
                    } else {
                        helpers.addResult(results, 2,
                            'Elastic Load Balancer does not have SSL Termination configured',
                            region, resource);
                    }
```

File: plugins/aws/elbv2/elbv2SslTermination.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            if (!describeLoadBalancers.data.length) {
+                helpers.addResult(results, 0, 'No Load Balancers found', region);
+                return rcb();
+            }
+
+            async.each(describeLoadBalancers.data, function(elb, cb){
+                
+                var resource = elb.LoadBalancerArn;
+
+                var describeListeners = helpers.addSource(cache, source,
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2SslTermination.js:40]:
```suggestion
            async.each(describeLoadBalancers.data, function(elb, cb){
                var resource = elb.LoadBalancerArn;

                var describeListeners = helpers.addSource(cache, source,
```

File: plugins/aws/elbv2/elbv2SslTermination.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                    return cb();
+                }
+
+                if (!describeListeners.data.Listeners || !describeListeners.data.Listeners.length){
+                    helpers.addResult(results, 2,
+                        'No Load Balancer listeners found',
+                        region, resource);
+                    return cb();
+                }
+
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2SslTermination.js:56]: let found = describeListeners.data.Listeners.find(listener => listener.Certificates && listener.Certificates.length);

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/elbv2/elbv2SslTermination.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                } else {
+                    helpers.addResult(results, 2,
+                        'Elastic Load Balancer does not have SSL Termination configured',
+                        region, resource);
+                }
+
+                cb();
+            }, function(){
+
+                rcb();
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2SslTermination.js:71]:
```suggestion
            }, function(){
                rcb();
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/elbv2/elbv2SslTermination.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'ELB SSL Termination',
+    category: 'ELBv2',
+    description: 'Ensure that Load Balancers has SSL certificate configured for SSL terminations.',
```
Comment (giorod3) [inline plugins/aws/elbv2/elbv2SslTermination.js:7]:
```suggestion
    description: 'Ensure that Load Balancers have SSL certificate configured for SSL terminations.',
```

**Review by giorod3 — APPROVED** [review]

**Pull request merged**